src/Package.js

```diff
@@ -60,12 +61,27 @@ export default class Package {
     return this._package.scripts || {};
   }
 
+  set versionSerializer(versionSerializer) {
+    const previousSerializer = this._versionSerializer;
+
+    this._versionSerializer = versionSerializer;
+
+    if (previousSerializer) {
+      this._package = previousSerializer.serialize(this._package);
+    }
+
+    if (versionSerializer) {
+      this._package = versionSerializer.deserialize(this._package);
+    }
+  }
+
   isPrivate() {
     return !!this._package.private;
   }
 
   toJSON() {
-    return this._package;
+    const pkg = _.cloneDeep(this._package);
```

Comment:
Can we move this `cloneDeep` into the version serializer methods? It seems overly defensive for the non-serialized case.

Reply:
We can, there's a couple of reasons why this is done like this:
- `toJSON` was returning its internal `pkg` object, any outside changes on it will affect Package
- cloning only for serializer case would make `toJSON` behave differently - in one case it will return its internal object the other time it would return a deep copy

Reply:
Despite `.toJSON()` _currently_ being used only in one place, I think you've raised some good points that will ultimately help future use cases (such as when I actually get around to refactoring lerna into a lerna-managed monorepo and the Package class is exposed in a package of its own). I was being overly-cautious about performance in a place where it really, really doesn't matter (package.json files aren't particularly large, and not deeply nested at all).

Thanks for the clear explanation. :)